### ISSUE TITLE
Send mail fails with "Forbidden"

### ISSUE THREAD

**Issue by ChuckBerg** [microsoftgraph/aspnet-connect-rest-sample#11]
I downloaded this sample code, registered it as an application in Azure Active Directory, gave it the required permissions - 'Sign in and read user profile' and 'Send mail as a user', and configured web.config with 'ClientID' and 'ClientSecret'.  The application allows me to connect with Office 365, but returns "Forbidden" when I attempt to send email.

Using the VS2015 debugger I captured the AccessToken that I receive upon login, and decoded it with http://jwt.calebb.net/.  It says the token provides "User.Read" permissions, and that's all.

Any thoughts on why I am not getting the permissions that I configured in AAD?

**Comment by DianeD** [comment]
Hi @ChuckBerg 
I registered a new app with those permissions and I'm able to successfully send mail. Mail.Send is one of the scopes in my token. Do you see the permission "Send mail as you" in the consent screen?

**Comment by ChuckBerg** [comment]
I don't even see a consent screen.  The sample code displays a "Connect to Office 365" button that takes me to login.microsoftonline.com.  After I provide credentials it takes me right back to the application.

**Comment by DianeD** [comment]
You should get the consent screen the first time you log in and use the app. You can revoke consent (click ... and then "Remove") from here and log in again to see if you get it. [https://myapps.microsoft.com](https://myapps.microsoft.com)

**Comment by ChuckBerg** [comment]
I went to that URL, clicked on "..." for the sample application.  I do not see "Remove" in the menu, but I do see "Get Info".  I clicked on that and it says the application has "Sign you in and read your profile" and "Send mail as you" permissions.

**Comment by ChuckBerg** [comment]
Hmmm, I'm starting to think there are two identities involved - one that the sample application is seeing and another that myapps is seeing.  That would explain why myapps doesn't have "Remove" in the menu - that identity has never been accessed by the sample app.  And it also explains why the sample app doesn't work - the identity it sees doesn't have the proper email permissions.

But how would that happen?

**Comment by ChuckBerg** [comment]
Here's what I did as near as I can remember.
1. I created an Office 365 developer account using our BizSpark subscription.  That automatically created a  ChuckBerg@****.onMicrosoft.com account.  I'm pretty sure this is a "Work Account" since it has the "badge" icon on the login.microsoftonline.com page.
2. Using the ChuckBerg@****.onMicrosoft.com credentials, I logged into the Office 365 administrator portal and created a bunch of mailboxes.
3. Using the ChuckBerg@****.onMicrosoft.com credentials, I logged into the Azure admin portal and launched the Azure Active Directory tool.  I was able to see all of the user accounts for which I had created Office 365 mailboxes.  This is where I registered the sample application.

OK, after drafting these steps, I logged into the Office 365 portal, found an opportunity to "Remove" permissions for the sample app, and did so.  Then ran the sample app again, approved access to send email, and the application worked!

Who knows!!!  I hope that automatic login by a service goes smoother!
